core/trino-main/src/main/java/io/trino/operator/PagesIndex.java

```diff
@@ -92,6 +93,7 @@
     private final IntArrayList positionCounts;
     private final boolean eagerCompact;
 
+    private int modificationCount; // may overflow, doesn't matter
```

Comment:
volatile?

Reply:
i don't think PagesIndex is supposed to be thread safe?